### PR TITLE
CPLAT-10914 Fix issues discovered during initial boilerplate rollout

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Motivation
There were some things I came across during rollout that I wanted to tweak the codemod to fix.

#### Related to FIXME comments:
- Fix issue with FIXME comments being added multiple times
- Don't put FIXME comments between node and annotation since it makes things harder to read
- Make the first sentence of FIXMEs that will remain in the code be all on one line, so they can be seen in an IDE's "TODO" view without going to the source
- Move FIXME instructions into a markdown file and link to that instead, so that code isn't as cluttered after running the codemod.

#### Related to codemod logic:
- Don't factor whether mixins have been converted into migration decision, since they're forwards compatible
- Generalize version constraint updating, add workiva_analysis_options to the list of dependencies that are added

#### Miscellaneous:
- Use build_vm_compilers so that tests load faster
- 6550f11af3442350a30b057b0dc303ee1d6b69b0 Fix component2 migrator choking on Dart `mixin` syntax in files (relevant since some consumers need to upgrade components to Component2 before upgrading to new boilerplate)

### Release Notes
- Miscellaneous boilerplate codemod fixes (see https://github.com/Workiva/over_react_codemod/pull/92 for more info)
- Fix Component2 codemod errors when encountering Dart `mixin`s

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/component2_suggestors/componentdidupdate_migrator.dart [hunk 1]
```diff
@@ -33,7 +33,8 @@ class ComponentDidUpdateMigrator extends GeneralizingAstVisitor
   visitMethodDeclaration(MethodDeclaration node) {
     super.visitMethodDeclaration(node);
 
-    ClassDeclaration containingClass = node.parent;
```
Comment (greglittlefield-wf) [inline lib/src/component2_suggestors/componentdidupdate_migrator.dart:36]: These were implicitly casting to `ClassDeclaration`, when the real type could be `ClassDeclaration`, `MixinDeclaration`, or `ExtensionDeclaration`.

File: test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart [hunk 2]
```diff
@@ -173,15 +173,11 @@ void advancedPropsAndStateClassMigratorTestHelper({
           expectedOutput: '''
```
Comment (greglittlefield-wf) [inline test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart:173]: The changes in this file should be just:
- FIXME comment updates
- Removal of tests related to mixins from external libs

**Comment by aaronlademann-wf** [comment]
QA +1

- [x] CI Passed

@Workiva/release-management-pp

**Pull request merged**